### PR TITLE
EKS Observability Accelerator -Nginx Module

### PR DESCRIPTION
**Pull request by saaish**
### More

- [ ] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [ ] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: examples/observability/eks-nginx/README.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+aws eks --region $AWS_REGION update-kubeconfig --name aws001-preprod-observability-eks
+```
+
+`terraform apply` will provision all the aforementioned resources.
+
+---
+
+**NOTE**
+
+This example deploy all the necessary components to start monitoring your Java-based
```
Comment (askulkarni2) [inline examples/observability/eks-nginx/README.md:89]: nginx based applications

File: examples/observability/eks-nginx/README.md [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+- 3. Enter the following commands to install HAProxy:
+```
+helm install my-nginx ingress-nginx/ingress-nginx \
+--namespace nginx-ingress-sample \
+--set controller.metrics.enabled=true \
+--set-string controller.metrics.service.annotations."prometheus\.io/port"="10254" \
+--set-string controller.metrics.service.annotations."prometheus\.io/scrape"="true"
+
+```
```
Comment (askulkarni2) [inline examples/observability/eks-nginx/README.md:148]: Can we use the existing ingress-nginx add-on in this example?

File: modules/kubernetes-addons/adot-collector-nginx/locals.tf [hunk 3]
```diff
@@ -0,0 +1,60 @@
+locals {
+  name                           = "adot-collector"
+  adot_collector_service_account = "adot-collector"
```
Comment (askulkarni2) [inline modules/kubernetes-addons/adot-collector-nginx/locals.tf:3]: This needs to be its own namespace and service-account `adot-collector-nginx`. We can consolidate this going forward.

File: modules/kubernetes-addons/adot-collector-nginx/otel-config/templates/clusterrolebinding.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: otel-prometheus-role-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: otel-prometheus-role
+subjects:
+  - kind: ServiceAccount
+    name: adot-collector
+    namespace: adot-collector
```
Comment (askulkarni2) [inline modules/kubernetes-addons/adot-collector-nginx/otel-config/templates/clusterrolebinding.yaml:12]: `adot-collector-nginx`, also add a newline after to pass linter checks.

File: modules/kubernetes-addons/variables.tf [hunk 5]
```diff
@@ -711,3 +711,15 @@ variable "adot_collector_java_helm_config" {
   default     = {}
   description = "ADOT Collector Java Helm Chart config"
 }
+
```
Comment (askulkarni2) [inline modules/kubernetes-addons/variables.tf:714]: Add a comment for separator. See others for format.

**Review by askulkarni2 — COMMENTED** [review]
Looking good. Some minor comments.

File: examples/observability/eks-nginx/README.md (hunk 1 above)
Comment (saaish) [reply examples/observability/eks-nginx/README.md]: Corrected

File: modules/kubernetes-addons/adot-collector-nginx/locals.tf (hunk 3 above)
Comment (saaish) [reply modules/kubernetes-addons/adot-collector-nginx/locals.tf:3]: Copy

File: modules/kubernetes-addons/adot-collector-nginx/otel-config/templates/clusterrolebinding.yaml (hunk 4 above)
Comment (saaish) [reply modules/kubernetes-addons/adot-collector-nginx/otel-config/templates/clusterrolebinding.yaml]: modified

**Review by askulkarni2 — APPROVED** [review]

**Review by bryantbiggs — APPROVED** [review]
Thank you @saaish !

**Pull request merged**